test/integration_test.rb

```diff
@@ -229,6 +235,46 @@ def test_mixed_borders
     end
     @workbook.apply_styles
     assert_equal 8, @workbook.styled_cells.count
-    assert_equal 6, @workbook.style_index.count
+    assert_equal 6, @workbook.styles.style_index.count
+  end
+
+  def test_merge_styles_1
+    @filename = 'merge_styles_1'
+    @workbook.add_worksheet do |sheet|
+      sheet.add_row
+      sheet.add_row ['', '1', '2', '3']
+      sheet.add_row ['', '4', '5', '6']
+      sheet.add_row ['', '7', '8', '9']
+      sheet.add_style 'B2:D4', b: true
+      sheet.add_style 'B2:D4', border: { style: :thin, color: '000000' }
+    end
+    @workbook.apply_styles
```

Comment:
Actually, I was wrong on my previous comment:
```
sheet.add_style 'B2:D4', border: { style: :thin, color: '000000' }
```
does exactly what it's supposed to do, which is apply border to every cell in the range. This is how `axlsx` works. The default border is a border applied to every edge of the cell.